### PR TITLE
Add serverless git auth integration tests

### PR DESCRIPTION
**Pull request by moelsayed**
**Description**

Changes proposed in this pull request:

- Add serverless git auth integration tests. 

This PR is a periodic job. So it works on main. The kyma PR needs to be merged first for the pjtester to make sense here.

**Related issue(s)**

https://github.com/kyma-project/kyma/issues/12521

### CODE REVIEW

File: templates/data/kyma-integration-data.yaml [hunk 1]
```diff
@@ -487,6 +487,27 @@ templates:
...
+                  annotations:
+                    description: Serverless git-function authentication nightly integration tests
+                    pipeline.platform: k3s
+                  labels:
+                    preset-build-main: "true"
+                    preset-serverless-git-auth: "true"
+                inheritedConfigs:
+                  global:
+                    - "jobConfig_default"
+                    - "jobConfig_presubmit"
```
Comment (Halamix2) [inline templates/data/kyma-integration-data.yaml:503]:
```suggestion
                    - "jobConfig_periodic"
```

File: prow/scripts/cluster-integration/serverless-git-auth-integration-nightly-k3s.sh [hunk 2]
```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env bash
```
Comment (dbadura) [inline prow/scripts/cluster-integration/serverless-git-auth-integration-nightly-k3s.sh:1]: Can we use one script `serverless-integration-k3s.sh` and pass parameter: e.g.: `job_name="k3s-serverless-nightly-test"?

**Review by dbadura — CHANGES_REQUESTED** [review]

File: prow/scripts/cluster-integration/serverless-git-auth-integration-nightly-k3s.sh (hunk 2 above)
Comment (moelsayed) [reply prow/scripts/cluster-integration/serverless-git-auth-integration-nightly-k3s.sh:1]: Initially I just added a few if conditions tot eh original script, but it got a bit ugly. So, I moved shared code an external file and split the tests.

Comment (dbadura) [reply prow/scripts/cluster-integration/serverless-git-auth-integration-nightly-k3s.sh:1]: Is there any difference (except env) running gitauth test and integration-tests?

File: prow/scripts/cluster-integration/serverless-integration-k3s.sh [hunk 3]
```diff
@@ -149,7 +26,12 @@ kyma provision k3d --ci
 
 echo "--> Deploying Serverless"
 # The python38 function requires 40M+ of memory to work. Mostly used by kubeless. I need to overrride the defaultPreset to M to avoid OOMkill.
-kyma deploy -p evaluation --component cluster-essentials,serverless --ci --value "$REGISTRY_VALUES" --value global.ingress.domainName="$DOMAIN" --value "serverless.webhook.values.function.resources.defaultPreset=M" -s local -w $KYMA_SOURCES_DIR
+kyma deploy -p evaluation --ci \
```
Comment (dbadura) [inline prow/scripts/cluster-integration/serverless-integration-k3s.sh:29]: Good change :)

**Review by dbadura — COMMENTED** [review]
Looks ok.

File: prow/scripts/cluster-integration/serverless-integration-k3s.sh [hunk 4]
```diff
@@ -166,11 +62,23 @@ sleep 60
...
+
+VALUES="-f ${SERVERLESS_CHART_DIR}/values.yaml"
+# check for test overrides.
+if [[ -e "${SERVERLESS_OVERRIDES_DIR}/integration-overrides.yaml" ]]; then
+  VALUES+=" -f ${SERVERLESS_OVERRIDES_DIR}/integration-overrides.yaml"
+fi
 
-helm install serverless-test "${SERVERLESS_CHART_DIR}/charts/k3s-tests" -n default -f "${SERVERLESS_CHART_DIR}/values.yaml" --set jobName="${job_name}"
+#shellcheck disable=SC2086
+helm install serverless-test "${SERVERLESS_CHART_DIR}/charts/k3s-tests" -n default ${VALUES} --set jobName="${job_name}"
```
Comment (dbadura) [inline prow/scripts/cluster-integration/serverless-integration-k3s.sh:79]: I retriggered pjtester, but I coulnd't find test `git-auth-integration`.
I only found: k3s: `serverless-integration` and `serverless-git-auth-integration` which run the same suite: `serverless-integration`, which is the default one.

I noticed that you didn't pass the override for testSuite while installing the chart.

**Review by dbadura — CHANGES_REQUESTED** [review]

File: prow/scripts/provision-vm-and-start-serverless-k3s.sh [hunk 5]
```diff
@@ -107,12 +107,29 @@ ENDTIME=$(date +%s)
 echo "VM creation time: $((ENDTIME - STARTTIME)) seconds."
 
 trap cleanup exit INT
+# apply overrides if we are not using the default test suite
+if [[ ${INTEGRATION_SUITE} != "serverless-integration" ]]; then
```
Comment (dbadura) [inline prow/scripts/provision-vm-and-start-serverless-k3s.sh:111]:
```suggestion
if [[ ${INTEGRATION_SUITE} == "git-auth-integration" ]]; then
```

File: prow/scripts/cluster-integration/serverless-integration-k3s.sh [hunk 6]
```diff
@@ -166,11 +62,23 @@ sleep 60
...
-helm install serverless-test "${SERVERLESS_CHART_DIR}/charts/k3s-tests" -n default -f "${SERVERLESS_CHART_DIR}/values.yaml" --set jobName="${job_name}"
+VALUES="-f ${SERVERLESS_CHART_DIR}/values.yaml"
+# check for test overrides.
+if [[ -e "${SERVERLESS_OVERRIDES_DIR}/integration-overrides.yaml" ]]; then
+  VALUES+=" -f ${SERVERLESS_OVERRIDES_DIR}/integration-overrides.yaml"
+fi
 
-job_status=""
+#shellcheck disable=SC2086
+helm install serverless-test "${SERVERLESS_CHART_DIR}/charts/k3s-tests" -n default ${VALUES} --set jobName="${job_name}"
```
Comment (dbadura) [inline prow/scripts/cluster-integration/serverless-integration-k3s.sh:79]:
```suggestion
helm install serverless-test "${SERVERLESS_CHART_DIR}/charts/k3s-tests" -n default ${VALUES} \
 --set jobName="${job_name}" \
 --set testSuite="${INTEGRATION_SUITE}"
```

**Review by dbadura — APPROVED** [review]

**Review by dbadura — DISMISSED** [review]

**Review by Halamix2 — DISMISSED** [review]

**Review by Halamix2 — APPROVED** [review]

**Review by dbadura — APPROVED** [review]

**Review by mmitoraj — APPROVED** [review]

**Pull request merged**